gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -64,12 +69,20 @@
   // Default GCS upload granularity.
   static final int GCS_MINIMUM_CHUNK_SIZE = 256 * 1024;
 
-  private static final Duration START_RESUMABLE_WRITE_TIMEOUT = Duration.ofMinutes(10);
-  private static final Duration QUERY_WRITE_STATUS_TIMEOUT = Duration.ofMinutes(10);
-  private static final Duration WRITE_STREAM_TIMEOUT = Duration.ofMinutes(20);
+  private static final Duration START_RESUMABLE_WRITE_TIMEOUT = Duration.ofSeconds(10);
+  private static final Duration QUERY_WRITE_STATUS_TIMEOUT = Duration.ofSeconds(10);
+  private static final Duration WRITE_STREAM_TIMEOUT = Duration.ofMinutes(10);
   // Maximum number of automatic retries for each data chunk
   // when writing to underlying channel raises error.
-  private static final int UPLOAD_RETRIES = 10;
+  private static final int UPLOAD_RETRIES = 5;
+
+  // Number of insert requests to retain, in case we need to rewind and resume an upload. Using too
+  // small of a number could risk being unable to resume the write if the resume point is an
+  // already-discarded buffer; and setting the value too high wastes RAM. Note: We could have a
+  // more complex implementation that periodically queries the service to find out the last
+  // committed offset, to determine what's safe to discard, but that would also impose a performance
+  // penalty.
+  private static int numberOfRequestsToRetain = 5;
```

Comment:
Convert name to capitalized snake case, because this is a constant.

Reply:
Done.